src/dmtcp_dlsym.cpp

```diff
@@ -545,12 +546,8 @@ print_debug_messages(dt_tag tags,
 EXTERNC void *
 dmtcp_dlsym(void *handle, const char *symbol)
 {
-  // Acquire checkpoint lock. We need to release it before returning it from the
-  // function. Ideally, we would use the WrapperLock class to avoid calling
-  // dmtcp_enable_ckpt multiple times within the same function but that would
-  // require removing this file's dependency from dmtcp_restart.cpp and
-  // adjusting Makefile.am.
-  dmtcp_disable_ckpt();
+  // Acquire checkpoint lock.
```

Comment:
Let's add to the comment something like `// wrapperExecutionLock is a local variable. C++ releases it on function return.`

This helps to avoid misleading students who are new to C++.  Otherwise, they see "Acquire", with no corresponding comment about "Release".